core/injector/application_injector.cpp

```diff
@@ -287,23 +288,52 @@ namespace {
       common::raise(trie_storage_res.error());
     }
 
-    const auto genesis_raw_configs = configuration_storage->getGenesis();
+    const auto &genesis_raw_configs =
+        configuration_storage->getGenesisTopSection();
     auto &trie_storage = trie_storage_res.value();
 
-    auto batch_res =
-        trie_storage->getPersistentBatchAt(serializer->getEmptyRootHash());
-    if (not batch_res) {
-      common::raise(batch_res.error());
-    }
-    auto batch = std::move(batch_res.value());
-
     auto log = log::createLogger("Injector", "injector");
-    for (const auto &[key_, val_] : genesis_raw_configs) {
-      auto &key = key_;
-      auto &val = val_;
-      SL_TRACE(
-          log, "Key: {}, Val: {}", key.toHex(), val.toHex().substr(0, 200));
-      if (auto res = batch->put(key, val); not res) {
+    auto batch_with_raw_data = [&log, &serializer, &trie_storage](
+                                   const auto &raw_configs) {
+      auto batch_res =
+          trie_storage->getPersistentBatchAt(serializer->getEmptyRootHash());
+      if (not batch_res) {
+        common::raise(batch_res.error());
+      }
+      auto batch = std::move(batch_res.value());
+
+      for (const auto &[key_, val_] : raw_configs) {
+        auto &key = key_;
+        auto &val = val_;
```

Comment:
?
```c++
for (const auto &[key, val] : raw_configs)
    ~~auto &key = key_;~~
    ~~auto &val = val_;~~

Reply:
Done

Reply:
I reverted it back, because it has compilation error on some builds and not affects my local build. An error smth like use of automatic storage variable inside nested lambda call. Actually I didn't write this code, I moved it.